DAQ_System/Source/main.cpp

```diff
@@ -112,117 +103,117 @@ int main()
     }
 
     printf("Closing the root directory... ");
     fflush(stdout);
-    err = closedir(d);
-    printf("%s\n", (err < 0 ? "Fail :(" : "OK"));
-    if (err < 0) {
+    status = closedir(root_dir);
+    printf("%s\n", (status < 0 ? "Fail :(" : "OK"));
+    if (status < 0) {
         error("error: %s (%d)\n", strerror(errno), -errno);
     }
 
     // Display the numbers file
     printf("Opening \"/fs/numbers.txt\"... ");
     fflush(stdout);
-    f = fopen("/fs/numbers.txt", "r");
-    printf("%s\n", (!f ? "Fail :(" : "OK"));
-    if (!f) {
+    numbers_file = fopen("/fs/numbers.txt", "r");
+    printf("%s\n", (!numbers_file ? "Fail :(" : "OK"));
+    if (!numbers_file) {
         error("error: %s (%d)\n", strerror(errno), -errno);
     }
 
     printf("numbers:\n");
-    while (!feof(f)) {
-        int c = fgetc(f);
-        printf("%c", c);
+    while (!feof(numbers_file)) {
+        int num = fgetc(numbers_file);
+        printf("%c", num);
     }
 
     printf("\rClosing \"/fs/numbers.txt\"... ");
     fflush(stdout);
-    err = fclose(f);
-    printf("%s\n", (err < 0 ? "Fail :(" : "OK"));
-    if (err < 0) {
+    status = fclose(numbers_file);
+    printf("%s\n", (status < 0 ? "Fail :(" : "OK"));
+    if (status < 0) {
         error("error: %s (%d)\n", strerror(errno), -errno);
     }
 
     // Tidy up
     printf("Unmounting... ");
     fflush(stdout);
-    err = fileSystem.unmount();
-    printf("%s\n", (err < 0 ? "Fail :(" : "OK"));
-    if (err < 0) {
-        error("error: %s (%d)\n", strerror(-err), err);
+    status = file_system.unmount();
+    printf("%s\n", (status < 0 ? "Fail :(" : "OK"));
+    if (status < 0) {
+        error("error: %s (%d)\n", strerror(-status), status);
     }
     
     printf("Initializing the block device... ");
     fflush(stdout);
 
-    err = blockDevice.init();
-    printf("%s\n", (err ? "Fail :(" : "OK"));
-    if (err) {
-        error("error: %s (%d)\n", strerror(-err), err);
+    status = block_device.init();
+    printf("%s\n", (status ? "Fail :(" : "OK"));
+    if (status) {
+        error("error: %s (%d)\n", strerror(-status), status);
     }
 
     printf("Erasing the block device... ");
     fflush(stdout);
-    err = blockDevice.erase(0, blockDevice.size());
-    printf("%s\n", (err ? "Fail :(" : "OK"));
-    if (err) {
-        error("error: %s (%d)\n", strerror(-err), err);
+    status = block_device.erase(0, block_device.size());
+    printf("%s\n", (status ? "Fail :(" : "OK"));
+    if (status) {
+        error("error: %s (%d)\n", strerror(-status), status);
     }
 
     printf("Deinitializing the block device... ");
     fflush(stdout);
-    err = blockDevice.deinit();
-    printf("%s\n", (err ? "Fail :(" : "OK"));
-    if (err) {
-        error("error: %s (%d)\n", strerror(-err), err);
+    status = block_device.deinit();
+    printf("%s\n", (status ? "Fail :(" : "OK"));
+    if (status) {
+        error("error: %s (%d)\n", strerror(-status), status);
```

Comment:
Looks like here's a brief demonstration of how to use the block device class. Let's leave this alone for now. We can support those features separately in the Platform layer. For now, let's maintain our focus on the data logging.